core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -330,6 +331,15 @@ private[spark] object PythonRDD extends Logging {
   }
 
   /**
+   * Return an RDD of values from an RDD of (Long, Array[Byte]), with preservePartitions=true
+   *
+   * This is useful for PySpark to have the partitioner after partitionBy()
+   */
+  def valueOfPair(pair: JavaPairRDD[Long, Array[Byte]]): JavaRDD[Array[Byte]] = {
```

Comment:
In Scala/Java API, RDD.values() will change the RDD from (K, V) into RDD of V, so `preservePartitions` should not be `true`.

For PySpark, it change the RDD from (hash, [(K, V)]) to (K, V), `preservePartitions` should be true.